### PR TITLE
basic structure of the gnrc-aodvv2 routing protocol

### PR DESCRIPTION
**Pull request by gustavosinbandera1**
very basic structure of the project, to be able to do simulations and compile gnrc_aodvv2 module

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,4 +15,70 @@ DEVELHELP ?= 1
...
 
+
+USEPKG += oonf_api
+USEMODULE += oonf_rfc5444 
+
+BOARD_PROVIDES_NETIF := acd52832 airfy-beacon atmega256rfr2-xpro avr-rss2 b-l072z-lrwan1 cc2538dk fox \
+        derfmega128 derfmega256 hamilton iotlab-m3 iotlab-a8-m3 lobaro-lorabox lsn50 mulle microbit msba2 \
+        microduino-corerf native nrf51dk nrf51dongle nrf52dk nrf52840dk nrf52840-mdk nrf6310 \
+        nucleo-f207zg nucleo-f767zi openmote-b openmote-cc2538 pba-d-01-kw2x remote-pa \
+        remote-reva ruuvitag samr21-xpro samr30-xpro spark-core telosb thingy52 yunjia-nrf51822 z1
```
Comment (jeandudey) [inline Makefile]: Should be removed

File: Makefile [hunk 2]
```diff
@@ -15,4 +15,70 @@ DEVELHELP ?= 1
...
+
+
+# use IEEE 802.15.4 as link-layer protocol
+ #USEMODULE += netdev_ieee802154
+ #USEMODULE += netdev_test
+
+# We use only the lower layers of the GNRC network stack, hence, we can
+# reduce the size of the packet buffer a bit
+CFLAGS += -DGNRC_PKTBUF_SIZE=512
+endif
```
Comment (jeandudey) [inline Makefile]: Remove `ifneq` condition and unindent the code.

File: Makefile [hunk 3]
```diff
@@ -15,4 +15,70 @@ DEVELHELP ?= 1
...
+
+USEMODULE += aodvv2
+
+
+
+export INCLUDES += -I${RIOTBASE}/sys/include/net/
+export INCLUDES += -I${RIOTBASE}/sys/include/net/gnrc/aodvv2/
+export INCLUDES += -I${RIOTBASE}/sys/net/gnrc/aodvv2/
```
Comment (jeandudey) [inline Makefile]: This shouldn't be here.

File: main.c [hunk 4]
```diff
@@ -10,14 +10,138 @@
...
-int main(void)
-{
-    puts("Hello World!");
+#define ENABLE_DEBUG (1)
+#include "debug.h"
+#include <inttypes.h>
 
-    printf("You are running RIOT on a(n) %s board.\n", RIOT_BOARD);
-    printf("This board features a(n) %s MCU.\n", RIOT_MCU);
+#include "aodvv2.h"
```
Comment (jeandudey) [inline main.c]: Change to `#include "net/gnrc/aodvv2/aodvv2.h"`

File: shell_functions.c [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+    DEBUG("tenemos una interface PID: %d\n", _mock_netif->pid);
+  }
+
+  void *iter_state = NULL;
+  gnrc_ipv6_nib_nc_t nce;
+  int res;
+  (void)res;
+
+  ipv6_addr_t neighbor;
+  (void)neighbor;
```
Comment (jeandudey) [inline shell_functions.c]: this line can be removed

File: main.c [hunk 6]
```diff
@@ -10,14 +10,138 @@
  * 
  */
 
-#include <stdio.h>
+// #include <stdio.h>
+// #include <stdlib.h>
+// #include <string.h>
```
Comment (jeandudey) [inline main.c]: Remove

File: main.c [hunk 7]
```diff
@@ -10,14 +10,138 @@
...
+#include <unistd.h> // for getting the pid
```
Comment (jeandudey) [inline main.c]: Remove pid comment, not necessary

File: main.c [hunk 8]
```diff
@@ -10,14 +10,138 @@
...
 
-    return 0;
+#include "msg.h"
+#include "shell.h"
+
+
+
+// #include <arpa/inet.h>
+// #include <netinet/in.h>
+// #include <sys/socket.h>
```
Comment (jeandudey) [inline main.c]: Remove

File: main.c [hunk 9]
```diff
@@ -10,14 +10,138 @@
...
+
+#include "eui64.h"
+#include "net/gnrc/ipv6/nib.h"
+
+#include "net/gnrc/ipv6/nib.h"
+#include "net/gnrc/ipv6/nib/nc.h"
+
+#include "net/gnrc.h"
+// #include "net/gnrc/pktdump.h"
```
Comment (jeandudey) [inline main.c]: Remove

File: main.c [hunk 10]
```diff
@@ -10,14 +10,138 @@
...
+#include "net/gnrc/netif/hdr.h"
+#include "net/gnrc/pktdump.h"
+#include "net/netdev_test.h"
+#include "xtimer.h"
+
+
+
+// static gnrc_netreg_entry_t _dumper;
+// static msg_t _dumper_queue[DUMPER_QUEUE_SIZE];
+// static char _dumper_stack[THREAD_STACKSIZE_MAIN];
```
Comment (jeandudey) [inline main.c]: Remove if not used

File: main.c [hunk 11]
```diff
@@ -10,14 +10,138 @@
...
+  if (_mock_netif != NULL) {
+    DEBUG("tenemos una interface PID: %d\n", _mock_netif->pid);
+    ipv6_addr_t ipv6_addr;
+    int res = gnrc_netapi_get(_mock_netif->pid, NETOPT_IPV6_ADDR, 0,
+                              (void *)&ipv6_addr, sizeof(ipv6_addr));
+
+    char addr_str[IPV6_ADDR_MAX_STR_LEN];
+     ipv6_addr_to_str(addr_str, &ipv6_addr, sizeof(ipv6_addr));
+    DEBUG("-------------------dst = %s",addr_str);
+    printf("***********la salida es %d\n", res);
```
Comment (jeandudey) [inline main.c]: Use either `printf` or `debug`, not both. Remove dashes and asterisks.

File: main.c [hunk 12]
```diff
@@ -10,14 +10,138 @@
...
+
+  msg_init_queue(msg_q, RCV_MSG_Q_SIZE);
+  init_socket();
+  aodv_init();
+
+
+  (void)puts("Welcome to RIOT!\n");
+  _mock_netif = gnrc_netif_iter(_mock_netif);
+  if (_mock_netif != NULL) {
+    DEBUG("tenemos una interface PID: %d\n", _mock_netif->pid);
```
Comment (jeandudey) [inline main.c]: English

File: main.c (hunk 11 above)
Comment (jeandudey) [reply main.c]: And English

File: shell_functions.c [hunk 13]
```diff
@@ -0,0 +1,264 @@
...
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h> // for getting the pid
+
+#define ENABLE_DEBUG (1)
+#include "debug.h"
+#include <inttypes.h>
+
+#include "routingtable.h"
```
Comment (jeandudey) [inline shell_functions.c]: -> `net/gnrc/aodvv2/routingtable.h`

File: shell_functions.c [hunk 14]
```diff
@@ -0,0 +1,264 @@
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h> // for getting the pid
```
Comment (jeandudey) [inline shell_functions.c]: Remove comment.

File: shell_functions.c [hunk 15]
```diff
@@ -0,0 +1,264 @@
...
+
+#define DUMPER_QUEUE_SIZE (16)
+// constants from the AODVv2 Draft, version 03
+#define DISCOVERY_ATTEMPTS_MAX (1) //(3)
+#define RREQ_WAIT_TIME (2000000) 
+char _rcv_stack_buf[THREAD_STACKSIZE_MAIN];
+
+#define NBR_MAC                                                                \
+  { 0x57, 0x44, 0x33, 0x22, 0x11, 0x00, }
+#define NBR_LINK_LOCAL                                                         \
```
Comment (jeandudey) [inline shell_functions.c]: Explain usage

File: shell_functions.c [hunk 16]
```diff
@@ -0,0 +1,264 @@
...
+#include "net/gnrc/ipv6/nib.h"
+#include "net/gnrc/ipv6/nib/nc.h"
+
+#define DUMPER_QUEUE_SIZE (16)
+// constants from the AODVv2 Draft, version 03
+#define DISCOVERY_ATTEMPTS_MAX (1) //(3)
+#define RREQ_WAIT_TIME (2000000) 
+char _rcv_stack_buf[THREAD_STACKSIZE_MAIN];
+
+#define NBR_MAC                                                                \
```
Comment (jeandudey) [inline shell_functions.c]: Explain usage

File: shell_functions.c [hunk 17]
```diff
@@ -0,0 +1,264 @@
...
+#define DST                                                                    \
+  {                                                                            \
+    0x20, 0x01, 0x0d, 0xb8, 0x00, 0x00, 0xab, 0xcd, 0x55, 0x44, 0x33, 0xff,    \
+        0xfe, 0x22, 0x11, 0x00,                                                \
+  }
+#define DST_PFX_LEN (64U)
+
+/* IPv6 header + payload:https://github.com/gustavosinbandera1/RIOT version+TC
+ * FL: 0       plen: 16    NH:17 HL:64 */
+#define L2_PAYLOAD                                                             \
```
Comment (jeandudey) [inline shell_functions.c]: Explain usage

File: shell_functions.c (hunk 15 above)
Comment (gustavosinbandera1) [reply shell_functions.c]: was just for testing

Comment (gustavosinbandera1) [reply shell_functions.c]: when I was trying to get and save neighbors, it can be removed

**Comment by jeandudey** [comment]
Closing for now, as this work is already on master.

**Pull request closed without merging**